tools/src/main/java/com/nedap/archie/flattener/ArchetypeHRIDMap.java

```diff
@@ -0,0 +1,46 @@
+package com.nedap.archie.flattener;
+
+import com.nedap.archie.aom.ArchetypeHRID;
+import com.github.zafarkhaja.semver.Version;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class ArchetypeHRIDMap<T> extends ConcurrentHashMap<ArchetypeHRID,T> {
+
+    public T getLatestVersion(String archetypeId) throws IllegalArgumentException {
+        return getLatestVersion(new ArchetypeHRID(archetypeId));
+    }
+
+    public T getLatestVersion(ArchetypeHRID archetypeHRID){
+
+        //Early return for fully defined version
+        if (archetypeHRID.getPatchVersion() != null) {
+            return this.get(archetypeHRID);
+        }
+
+        //Filter if version is not fully defined
+        List<ArchetypeHRID> keys = this.keySet().stream().
+//                filter(id -> (id.getNamespace() == null && archetypeHRID.getNamespace() == null) ||
```

Comment:
Perhaps remove this comment after we decide to keep the getIdUpToConcept() method :)

Reply:
Removed comment.